src/main/java/infinity/parser/Parser.java

```diff
@@ -0,0 +1,102 @@
+package infinity.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.format.TextStyle;
+import java.util.Locale;
+
+/**
+ * This class handles the parsing of user inputs.
+ */
+public class Parser {
+
+    /**
+     * Checks the length of the value given whether it is within the specified range.
+     * Also checks if the value is parsable to an integer.
+     *
+     * @param value The value to be checked as a String.
+     * @param min The minimum length of the value.
+     * @param max The maximum length of the value.
+     * @param minNo The minimum value of the number.
+     * @param maxNo The maximum value of the number.
+     * @return True if the value is within the specified range and is a number, false otherwise.
+     */
+    private static boolean isNumberInRange(String value, int min, int max, int minNo, int maxNo) {
```

Comment:
Perhaps you could split this into multiple lines for clarity, rather than having one long return statement?